### ISSUE TITLE
how to implement rule only lint ts script in vue file?

### ISSUE THREAD

**Issue by tjx666** [vuejs/eslint-plugin-vue#2257]
I'm writing a eslint rule to forbid implict any variable declaration. This is my implementation:

```js
'use strict';

const MESSAGE_ID_DEFAULT = 'default';

/** @type {RuleCreate} */
const create = (context) => {
    return {
        VariableDeclaration(node) {
            // Only interested in let declarations without explicit type annotations
            if (node.kind === 'let') {
                for (const declarator of node.declarations) {
                    if (declarator.init === null && declarator.id.typeAnnotation == null) {
                        context.report({
                            node: declarator,
                            messageId: MESSAGE_ID_DEFAULT,
                        });
                    }
                }
            }
        },
    };
};

/** @type {RuleModule} */
module.exports = {
    meta: {
        type: 'suggestion',
        schema: [],
        messages: {
            [MESSAGE_ID_DEFAULT]: '禁止隐式声明类型为 any 的变量',
        },
    },
    create,
};
```

Then I configure it like:

```
{
	overrides: {
        {
            files: ['*.{ts,tsx}', '*.vue'],
            rules: {
                'design/no-declare-implicit-any-var': error,
            },
        },
    }
}
```

### Fail

```typescript
let resp;
try {
    resp = get()
} catch (error) {
    return null
}
```

### Pass

```typescript
let resp: any;
try {
    resp = get()
} catch (error) {
    return null
}
```

But this rule will also lint js script tag in vue file, I only want to lint ts file, any suggestion here?

related issue: https://github.com/vuejs/eslint-plugin-vue/issues/2217

<img width="746" alt="image" src="https://github.com/vuejs/eslint-plugin-vue/assets/41773861/94f1fd22-d71c-48d6-b4f0-2c6bae66757b">

**Comment by FloEdelmann** [comment]
See https://github.com/vuejs/eslint-plugin-vue/blob/master/lib/rules/require-typed-ref.js, that rule only reports errors in `.ts`, `.cts`, `.mts` file and in TypeScript SFCs.

**Comment by tjx666** [comment]
@FloEdelmann Seems can't deal with .vue with lang=ts but no setup
